tribits/core/package_arch/TribitsLibraryMacros.cmake

```diff
@@ -988,12 +988,14 @@ function(tribits_add_library LIBRARY_NAME_IN)
     # rest of the include directories from upstream packages should come from
     # the targets that have their INTERFACE_INCLUDE_DIRECTORIES property set.
 
-    #
     # Add ALIAS library <PackageName>::<libname>
-    #
-
     add_library(${PACKAGE_NAME}::${LIBRARY_NAME} ALIAS ${LIBRARY_NAME})
 
+    # Optionally Set IMPORTED_NO_SYSTEM
+    if (${PROJECT_NAME}_IMPORTED_NO_SYSTEM)
+      set_target_properties(${LIBRARY_NAME} PROPERTIES IMPORTED_NO_SYSTEM TRUE)
```

Comment:
Do we need to do the contrary in case of `if (! ${PROJECT_NAME}_IMPORTED_NO_SYSTEM)` ? I don't think since it is the default value. But that could also be a way to ensure that we know exactly the value of this property.

Reply:
No, I think it is better to leave a property at its default value if you don't set it to a non-default value that would change behavior.  Otherwise, we would be setting all kinds of properties to their default value in other such cases which would just confuse developers trying to inspect the set properties and debug things.  There is no recommendation to set properties to their default values in "Professional CMake: 10th Edition" (see Section "9.8. Recommended Practices" in "Chapter 9. Properties").

Reply:
Ok, I understand